js/viz/core/center_template.js

```diff
@@ -0,0 +1,86 @@
+
+import { patchFontOptions } from '../core/utils';
+import { noop } from '../../core/utils/common';
+
+const pieChartPlugin = {
+    name: 'center_template_pie_chart',
+    init: noop,
+    dispose: function() {
+        this._centerTemplateGroup.linkOff().dispose();
+    },
+    extenders: {
+        _createHtmlStructure() {
+            this._centerTemplateGroup = this._renderer.g()
+                .attr({ class: 'dxc-hole-template' })
+                .linkOn(this._renderer.root, 'center-template')
+                .css(patchFontOptions(this._themeManager._font))
+                .linkAppend();
+        },
+        _renderExtraElements() {
+            this._requestChange(['CENTER_TEMPLATE']);
+        },
+    },
+    members: {
+        _renderCenterTemplate() {
+            const template = this.option('centerTemplate');
+            const centerTemplateGroup = this._centerTemplateGroup.clear();
+
+            if(!template) {
+                return;
+            }
+            centerTemplateGroup.attr({ visibility: 'hidden' });
+
+            const center = this._getCenter();
+
+            this._getTemplate(template).render({
+                model: this,
+                container: centerTemplateGroup.element,
+                onRendered: () => {
+                    const group = centerTemplateGroup;
+                    const bBox = group.getBBox();
+                    group.move(center.x - (bBox.x + bBox.width / 2), center.y - (bBox.y + bBox.height / 2));
```

Comment:
Let's move part of calculations to separate variables to simplify this row?
```suggestion
                    const bBoxCenterX = bBox.x + bBox.width / 2;
                    const bBoxCenterY = bBox.y + bBox.height / 2;
                    group.move(center.x - bBoxCenterX, center.y - bBoxCenterY);
```